dotcom-rendering/src/components/ShowHideContainers.importable.tsx

```diff
@@ -45,9 +53,23 @@ export const ShowHideContainers = () => {
 			storage.local.set(`gu.prefs.container-states`, containerStates);
 		};
 
-		for (const e of window.document.querySelectorAll<HTMLElement>(
-			'[data-show-hide-button]',
-		)) {
+		const allShowHideButtons = Array.from(
+			window.document.querySelectorAll<HTMLElement>(
+				'[data-show-hide-button]',
+			),
+		);
+
+		const allContainersExpanded = allShowHideButtons.every(
```

Comment:
This will currently always return true as the attribute has not yet been toggled for containers that have been previously hidden by the user - this happens later in the call to `toggleContainer()` on line 79. This means that the show/hide buttons will be removed for all users, even if they have some containers hidden.

I think instead you may need to check whether `containerStates` has any closed containers at this point